### ISSUE TITLE
linter: `import/export` false positive when `export *` re-exports a `.tsx` module via a `.jsx` specifier

### ISSUE THREAD

**Issue by issac-lezon** [oxc-project/oxc#25951]
### What version of Oxlint are you using?

1.79.0

### What command did you run?

oxlint --config .oxlintrc.json .

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

{
  "categories": {
    "correctness": "off"
  },
  "plugins": ["import"],
  "rules": {
    "import/export": "error"
  }
}

### What happened?

`import/export` reports `No named exports found` on a parent `export *` barrel whose target only re-exports a `.tsx` module via a TypeScript ESM `.jsx` specifier.

```ts
// fail/leaf/Button.tsx
export const Button = () => <button />;

// fail/leaf/index.ts
export * from './Button.jsx';

// fail/index.ts
export * from './leaf/index.js';
```

```text
fail/index.ts:1:15: error import(export): No named exports found in module './leaf/index.js'
help: Remove the `export *` re-export, or add named exports to the target module.
```

TypeScript (`moduleResolution: "bundler"`) resolves `./Button.jsx` to `Button.tsx`, so `tsc --noEmit` accepts this tree.

This is not a barrel/`export *` bug and not a `.jsx` parse bug. The same tree passes if:

- the leaf specifier is `./Button.tsx`, or
- the leaf file is actually `Button.jsx`

The diagnostic does not appear on the leaf (`export * from './Button.jsx'`). Unresolved specifiers are skipped. It only appears on a parent that `export *`s that leaf.

The same false positive happens for `.js` → `.tsx` (`export * from './Button.js'` when the file is `Button.tsx`).

Likely cause: oxlint’s resolver `extension_alias` in `crates/oxc_linter/src/service/runtime.rs` (`get_resolver`) currently maps:

```rust
(".js",  [".js",  ".ts"])
(".mjs", [".mjs", ".mts"])
(".cjs", [".cjs", ".cts"])
```

TypeScript file-extension substitution also includes JSX (`.js` → `.tsx` / `.jsx`, `.jsx` → `.tsx`):

https://www.typescriptlang.org/docs/handbook/modules/reference.html#file-extension-substitution

Reproduction: `pnpm install && pnpm lint` (attach `oxlint-jsx-tsx-reexport.zip`).

[oxlint-jsx-tsx-reexport.zip](https://github.com/user-attachments/files/31294017/oxlint-jsx-tsx-reexport.zip)
